### ISSUE TITLE
Make a new logo

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#391]
Taken from FakeItEasy/fakeiteasy.github.io#5, which was repurposed to "use new logo".

As fond as we are of "two ladies talking", we don't really have a logo. Would be nice to have one in place for the 2.0 launch.

My understanding is that @philippdolder and BBV had started looking into it. @adamralph has indicated that we also has a contact that might be able to help. We've already seen my best efforts, and well, that's why we're asking professionals.

----

**update by @adamralph**

- [x] Send a PR to the website which adds a 128 wide logo image (just adding the file, no other changes) (@adamralph)
- [x] Once that is merged, send a PR to update the nuspec to point to the new logo URL (@adamralph)
- [x] Edit the 2.0 beta package on NuGet to point to the new logo URL (and optionally, the unlisted ones) (@adamralph)

----

**update by @adamralph**

- [x] switch 2.0 beta package to square logo
- [x] switch nuspec to square logo - https://github.com/FakeItEasy/FakeItEasy/pull/450

**Comment by adamralph** [comment]
Work is happening in the background on this. Expect to see some ideas soon.

**Comment by adamralph** [comment]
So it looks like we've homed in on a logo and favicons for 2.0. How shall we proceed?

How about the following:

* Send a PR to the website which adds a 128 wide logo image (just adding the file, no other changes)
* Once that is merged, send a PR to update the nuspec to point to the new logo URL
* Edit the 2.0 beta package on NuGet to point to the new logo URL (and optionally, the unlisted ones)
* Send a PR to the website to use the new logo and favicons, and any other reformatting/recolouring to make it fit better with the new logo
* Merge the website PR when we perform the 2.0 release

That way, our beta packages can use the logo ahead of time, to give early adopters a sneak peek and potentially drive out any issues, but the only the actual release contains the full make over.

**Comment by blairconrad** [comment]
That plan sounds good to me.

I wonder also about putting assets in the main repo, for example whatever
source format the logo and favicon will be in. Or did your third point mean
that you wanted to put our high quality images in the website repo as well
as whatever assets the website will be using?

  B

**Comment by adamralph** [comment]
I don't think we should put the source assets in the main repo. We could put them in a new repo though, e.g. https://github.com/NancyFx/Nancy.Portfolio

Come to think of it, we may want to use a larger image for the NuGet package since (I guess) high DPI scaled displays will actually use it, even though the element in nuget.org is sized to 128x128.

So, my idea for now is to add the following the website:

* `/images/logo_256.png` for use by the NuGet package
* `/images/logo_1024.png` for use in the website itself
* the favicon explosion generated by http://realfavicongenerator.net/

**Comment by blairconrad** [comment]
No objection here.

**Comment by adamralph** [comment]
First PR is here https://github.com/FakeItEasy/fakeiteasy.github.io/pull/8

**Comment by blairconrad** [comment]
Merged!

And I couldn't resist taking 5 minutes for:
[image: Inline image 1]

**Comment by adamralph** [comment]
5 minutes for what? I'm on tenterhooks...

**Comment by blairconrad** [comment]
Yeek! The image didn't come through in e-mail reply.

This wasn't a serious proposal. Just play.

![image](https://cloud.githubusercontent.com/assets/3275797/5989144/896b9b58-a94b-11e4-96d5-30c592d1a47d.png)

**Comment by blairconrad** [comment]
Although I did enjoy using the FAKE colours for the buttons in the footer.

**Comment by adamralph** [comment]
Play or not, I really like it! I think the button text would have to change to white, and perhaps we should remove the 'FakeItEasy' title in favour of just the logo. The footer would need to switch to white too.

**Comment by blairconrad** [comment]
Yeah, I'd played with the button text being white as well. Lemme do what you said, just for fun. I'm waiting for my humidifier to descale before I call my grandmother anyhow. If you know what I mean.

**Comment by blairconrad** [comment]
![image](https://cloud.githubusercontent.com/assets/3275797/5989261/5eeb2c82-a94f-11e4-9ba7-8e7e5589c205.png)

**Comment by blairconrad** [comment]
Hmm. Logo should prolly be shifted right a bit to align with the code block.

**Comment by adamralph** [comment]
Looks great!

I wonder if it might be better with the tag line immediately following the logo (bottom aligned), instead of being over on the right.

**Comment by blairconrad** [comment]
@adamralph, I'm happy to complete the last item in the checklist, but I don't think I have a final for the favicon.
Have I just missed it? If so, could you point me?
If not, want to send it or a PR or a new repo or…?

**Comment by adamralph** [comment]
I'll get it uploaded to a new repo. What shall we call it? fakeiteasy-assets?

**Comment by blairconrad** [comment]
Perfect.

**Comment by adamralph** [comment]
It's alive - https://github.com/FakeItEasy/fakeiteasy-assets

**Comment by adamralph** [comment]
At the moment it only contains the PDF sources. To create the PNG's, I exported them, choosing 600 DPI and transparent background, which results in a PNG about 4,000 pixels wide.

You may want to choose a lower DPI to get a smaller image for http://realfavicongenerator.net/. I also remember having to expand the canvas in paint.net to make the PNG square.

**Comment by blairconrad** [comment]
Thanks for putting the assets up there.

I'll work on the favicon in a bit. Yeah, 4000 pixels may be too big. I'm sure I can find an acceptable middle ground for it.

**Comment by adamralph** [comment]
You just need http://realfavicongenerator.net/ to accept it. The image that you feed into it is throw away.

**Comment by blairconrad** [comment]
Yup.

**Comment by blairconrad** [comment]
When tweets point at the NuGet package with the new logo, they look a little funny.

![image](https://cloud.githubusercontent.com/assets/3275797/5994410/473eb1fc-aa40-11e4-8446-fd04c93b1c83.png)

Hazard of a non-square logo, I would guess.

**Comment by adamralph** [comment]
Hmm, perhaps we should publish a variant with square padding for use in
NuGet and other sites which prefer it

**Comment by blairconrad** [comment]
Any objection to changing the main NuGet logo? (Not the packages'.) Twitter's already been done...

**Comment by adamralph** [comment]
What do you mean by 'main' NuGet logo? I only see the ones against packages.

**Comment by blairconrad** [comment]
I meant this

![image](https://cloud.githubusercontent.com/assets/3275797/6028342/c3f1e134-abb6-11e4-9bbc-57e01a5e11e6.png)

**Comment by blairconrad** [comment]
at https://www.nuget.org/packages/FakeItEasy/

**Comment by blairconrad** [comment]
Fixed.

**Comment by adamralph** [comment]
Oh, you changed the 1.x package logo to the new one? I thought we were reserving it for 2.x. Well, I guess it's fine...

**Comment by adamralph** [comment]
Hmmm... we may have to use a white background logo for Twitter. And perhaps other things...

![image](https://cloud.githubusercontent.com/assets/677704/6031462/eb7a33ac-abff-11e4-9466-0dd5722cc45f.png)

(from dark-themed TweetDeck)

**Comment by blairconrad** [comment]
Ah. No. That was a mistake. I thought there was a logo for "FakeItEasy" and one per package. I see that what I thought was the "FakeItEasy" logo was 1.25.2's. Reverted.

:sheep:

**Comment by blairconrad** [comment]
Yeah, that does not look so good on the dark background. Perhaps you should shift to light themed. :grinning: 

I do enjoy a transparent png, but sometimes being able to control your environment is really the way to go.

Translucent? Lemme run an experiment. No. That was terrible. White background is better.

Imagine the rounded corners

![image](https://cloud.githubusercontent.com/assets/3275797/6033351/9fea8df0-abe2-11e4-9b26-9613771f6e86.png)

**Comment by blairconrad** [comment]
I'm making a batch of whites right now. Will send a PR to fakeiteasy.github.io.

**Comment by adamralph** [comment]
I guess we may have pad them as well. That looks quite ugly with the logo hitting the walls.

**Comment by adamralph** [comment]
I've merged the white PR, but I'm wondering if we now need _padded variants of all as well :unamused:.

**Comment by adamralph** [comment]
This is how it appears right now

![image](https://cloud.githubusercontent.com/assets/677704/6040715/43e14cea-ac75-11e4-872b-fc10fab5c759.png)

**Comment by blairconrad** [comment]
I am happy to make logos all day long. (That was an actual serious comment. Love it.)

If we're going to have padding, though, I wonder if we need both padded and unpadded versions. Especially for the ones with opaque backgrounds. For me, if we see a need for a padded, white-backgrounded (square) logo, we might as well use it every where we want a white (square) logo.

**Comment by adamralph** [comment]
OK, let's just pad all the white background ones then, that probably makes sense.

**Comment by blairconrad** [comment]
Here's a 5% margin on the 256 pixel square:

![image](https://cloud.githubusercontent.com/assets/3275797/6046814/591fa2f2-ac73-11e4-8007-f9f283fb975e.png)

**Comment by adamralph** [comment]
I'd probably go bit more. How does 10% look?

I can see from that image that something is weird with the white fill. I've been using https://chocolatey.org/packages/PDFXChangeViewer to create the PNG's.

**Comment by blairconrad** [comment]
Sorry, you see something weird with the white fill on the new image I just posted?
Could you point it out?

10%:

![image](https://cloud.githubusercontent.com/assets/3275797/6047009/ecf834b6-ac74-11e4-9505-a0770e0134be.png)

**Comment by blairconrad** [comment]
Or maybe I should attach rather than taking a screenshot

![fakeiteasy_logo_256_square_white](https://cloud.githubusercontent.com/assets/3275797/6047018/057d9c56-ac75-11e4-8580-27c89d1eeed8.png)

**Comment by adamralph** [comment]
I think 10% looks good.

I see some very obvious artifacts on the borders of the T, for instance. Ultimately, I think the background needs to be applied on the vector source. A flood fill on a PNG will probably always result in some artifacts. If you like, I can export a high res padded white PNG using the vectors and send it to you?

**Comment by blairconrad** [comment]
Ah. I think I see what you're talking about.
The weird thing is that I'm not flood filling any more. I put the 2048 px  icon on top of a white layer, then flattened. Should've made a pretty image.

I don't think it's the layering. It's the downsampling method. PIL's ANTIALIAS method doesn't seem to be doing a great job, nor am I thrilled with bilinear or bicubic as offered by either PIL or Paint.NET.

Directly exporting at the desired size may be the only way to go.

**Comment by blairconrad** [comment]
This issue has been fixed in [release 2.0.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/2.0.0).
